src/i_winmusic.c

```diff
@@ -93,8 +96,9 @@ static UINT MidiDevice;
 static HMIDISTRM hMidiStream;
 static MIDIHDR MidiStreamHdr;
 static HANDLE hBufferReturnEvent;
-static HANDLE hExitEvent;
+static HANDLE hStoppedEvent;
 static HANDLE hPlayerThread;
+static CRITICAL_SECTION CriticalSection;
```

Comment:
Uh, no. First of all, wrong file. This is the right one: https://github.com/libsdl-org/SDL/blob/SDL2/src/thread/windows/SDL_sysmutex_c.h
Second, that uses the newer `SRWLOCK` https://learn.microsoft.com/en-us/windows/win32/sync/slim-reader-writer--srw--locks which is (supposedly) more efficient but requires a newer OS.

Reply:
Let's see, there's only two threads (main and midi) and contention rarely occurs (changing songs, shutting down, moving the volume slider). Do the performance benefits of srwlocks apply here? Is Chocolate Doom dropping WinXP support completely or simply no longer providing builds in favor of more convenient workflows?

(WinXP is dead and dangerous but has advantages when paired with offline retro systems. And it still has superior midi support: WinXP MS GS Synth has a higher polyphony and is better at handling missing instruments, various software/hardware midi devices don't work on Vista+, and the default midi device can be changed without third party hacks.)

Reply:
Also the old one is recursive (and I think can't be made non-recursive?) and the new one is not to better match POSIX semantics. But for us it does not really matter. Neither should efficiency. The only point is if we should make the decision ourselves or hand it off to SDL.

And I don't think we officially support XP anymore.

Reply:
> And I don't think we officially support XP anymore.

It depends on how we build a new release. The current http://latest.chocolate-doom.org/ still runs on WinXP. If we build a release using MSYS2 on GitHub CI, it will be Win7+.

Personally, I prefer to use native APIs in non-portable code.

Reply:
Let's keep the native API.